src/Infiltration/ui/SlashGame.tsx

```diff
@@ -27,34 +27,52 @@ const difficulties: {
 
 export function SlashGame({ difficulty, onSuccess, onFailure }: IMinigameProps): React.ReactElement {
   const [phase, setPhase] = useState(0);
-  const [hasAugment, setHasAugment] = useState(false);
-  const [guardingTime, setGuardingTime] = useState(0);
+  const timeOutId = useRef<number | ReturnType<typeof setTimeout>>(-1);
 
-  useEffect(() => {
-    // Determine timeframes for game phase changes
+  const data = useMemo(() => {
+    // Determine time window of phases
     const newDifficulty: Difficulty = { window: 0 };
     interpolate(difficulties, difficulty, newDifficulty);
     const distractedTime =
       newDifficulty.window * (Player.hasAugmentation(AugmentationName.WKSharmonizer, true) ? 1.3 : 1);
     const alertedTime = 250;
     const guardingTime = Math.random() * 3250 + 1500 - (distractedTime + alertedTime);
 
-    // Set initial game state
-    setPhase(0);
-    setGuardingTime(guardingTime);
-    setHasAugment(Player.hasAugmentation(AugmentationName.MightOfAres, true));
+    return {
+      hasAugment: Player.hasAugmentation(AugmentationName.MightOfAres, true),
+      guardingTime,
+      distractedTime,
+      alertedTime,
+    };
+  }, [difficulty]);
```

Comment:
You're memoizing the augment state here... isn't it possible for it to change without this component getting re-rendered as a result?